src/message_log.rs

```diff
@@ -216,6 +217,21 @@ impl PbftLog {
             return;
         }
 
+        // Except for Checkpoints and ViewChanges, the message must be for the current view to be
+        // accepted
+        let msg_type = PbftMessageType::from(msg.info().get_msg_type());
+        if msg_type != PbftMessageType::Checkpoint
+            && msg_type != PbftMessageType::ViewChange
+            && msg.info().get_view() != state.view
+        {
+            warn!(
+                "Not adding message with view number {}; does not match node's view: {}",
+                msg.info().get_view(),
+                state.view,
+            );
+            return;
```

Comment:
Should we be adding this message to the backlog instead of just not adding it to the set of messages? Alternatively, we might want to return a `Result` for whether or not the message was actually added, since over in `on_peer_message` it doesn't seem like we'd want to continue processing the message if we haven't added it to the log.

Reply:
I don't believe adding the message to the backlog is necessary; the only scenario in which it could be possible for a legitimate message to be rejected here would be if a node somehow missed a view change, so the other nodes are on the same sequence number but the next view. In this scenario the node can't do anything and it's just faulty, because it can't update its view until it has some proof of the view change; currently the only proof of a view change comes from the consensus seal, but we could possibly implement some other proof (say if we get 2f+1 matching `Prepare`s with a different view). I think this scenario is extremely unlikely though, so it's probably not worth worrying about unless we see otherwise.

As far as returning a result to stop processing the message, this is done in another PR that will follow.